lifecycle.go

```diff
@@ -62,62 +62,35 @@ func (l *lifecycle) Append(hook Hook) {
 	l.hooks = append(l.hooks, hook)
 }
 
-// start calls all OnStarts in order, halting on the first OnStart that fails
-// and marking that position so that stop can rollback.
-func (l *lifecycle) start() error {
+func (l *lifecycle) Start() error {
```

Comment:
I don't think this change is desirable.

The reason @breerly and I made it private to begin with is so module authors don't feel compelled to *accidentally* start/stop a lifecycle, which will obviously have a catastrophic effect on the service.

Apart from trusting module authors to do the right thing, I'd like to *completely* take away the possibility of that happening, if that makes sense.

Is this change required to move the test lifecycle to another package?